### DISCUSSION TITLE
Quarto book pages accumulate in file size

### DISCUSSION THREAD

**Discussion by SES-CE** [quarto-dev/quarto-cli#6107] [Q&A] [answered]
### Description

Hi

**Why do the HTML pages of my book accumulate in file size?** 

I am building an HTML book with Quarto. 

Recently I have run into issues with rendering the entire book. I get the error codes like: 

`pandoc: osCommitMemory: VirtualAlloc MEM_COMMIT failed to commit.......`

Which lead me to this [post](https://github.com/DS4PS/cpp-526-fall-2020/issues/25). I gather that it is a memory issue, so I tried to reduce the amount of stuff on my pages, i.e. remove unnecessary plots and shorten data frames. 

It did not fix the issue when rendering the book and that's where I noticed the file size of subsequent pages kept increasing, see snip below:  
![image](https://github.com/quarto-dev/quarto-cli/assets/88316918/7949ab8c-ca29-4ede-8203-5779b74f49b7)

Can it be a `embed-reasources` issue? 
I use HTML elements, plotly plots, so the header of each page looks like this:  

`---
title: "Title"
format:
  html:
    toc: true
    code-fold: true
    code-summary: "Show the code"
    embed-resources: true 
    css: theme.css

execute: 
  echo: false
`---
`

Thank you for your assistance with this.

**Comment by cscheid** [comment]
> Can it be a `embed-reasources` issue?

Yes. unfortunately, when `embed-resources` is true, Pandoc will load all of the necessary resources (javascript includes, css, and data) in main memory (it's not something we can control). In that case, the amount of memory allocated will be proportional to size of the final document, and will in general be significantly larger than that.

**Comment by cderv** [comment, marked as answer]
Can you share more also about why you are using `embed-resource` for a Quarto Book ? 
A HTML Quarto book won't be a single file document at the end, so usually it is ok to have a folder as a results, including HTML pages and some resources directories. 

Also, can you confirm that deactivating with `embed-resources: false` solve the size increment increase you are seeing ? thanks !

**Reply by SES-CE** [reply]
IIRC I had problems with HTML elements not showing when I rendered as a book vs. a single page. 
Now it works flawlessly with `embed-resources: false`! 

I **confirm** that it solves the incremental size increase.

Thanks for your patience and support.

**Comment by rreece** [comment]
I confirm that my output html files keep increasing in size when i compile as book, nearly 100 MB.
